boards.txt

```diff
@@ -590,6 +590,32 @@ GenF4.menu.pnum.BLACK_F407VE.build.product_line=STM32F407xx
 GenF4.menu.pnum.BLACK_F407VE.build.variant=BLACK_F407VE
 GenF4.menu.pnum.BLACK_F407VE.build.cmsis_lib_gcc=arm_cortexM4l_math
 
+# Black F407ZE
+# Support: Serial1 (USART1 on PA10, PA9)
+# Default SPI: SPI (SPI1 on PB3,PB4,PB5 - NRF24 connector and W25Q16 (on board flash))
+GenF4.menu.pnum.BLACK_F407ZE=Black F407ZE
+GenF4.menu.pnum.BLACK_F407ZE.upload.maximum_size=524288
+GenF4.menu.pnum.BLACK_F407ZE.upload.maximum_data_size=131072
+GenF4.menu.pnum.BLACK_F407ZE.build.mcu=cortex-m4 -mfpu=fpv4-sp-d16 -mfloat-abi=hard
+GenF4.menu.pnum.BLACK_F407ZE.build.board=BLACK_F407ZE
+GenF4.menu.pnum.BLACK_F407ZE.build.series=STM32F4xx
+GenF4.menu.pnum.BLACK_F407ZE.build.product_line=STM32F407xx
+GenF4.menu.pnum.BLACK_F407ZE.build.variant=BLACK_F407VE
+GenF4.menu.pnum.BLACK_F407ZE.build.cmsis_lib_gcc=arm_cortexM4l_math
+
+# Black F407ZG
+# Support: Serial1 (USART1 on PA10, PA9)
+# Default SPI: SPI (SPI1 on PB3,PB4,PB5 - NRF24 connector and W25Q16 (on board flash))
+GenF4.menu.pnum.BLACK_F407ZG=Black F407ZG
+GenF4.menu.pnum.BLACK_F407ZG.upload.maximum_size=1048576
```

Comment:
As more Flash, ldscript have also to be updated using switch for FLASH definition

Reply:
Hmm, I'm not sure how to do a conditional in a linker script, does the ZGT need breaking out into another variant folder?

Reply:
Simply like this https://github.com/ARMmbed/mbed-os/blob/90fd6cdf27cc11b581627ace42ec10ae77ccf0f0/targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L432xC/device/TOOLCHAIN_GCC_ARM/STM32L432XX.ld#L5

Reply:
I think this only works if the linker script is pre-processed, it does not appear to be so - ld throws an exception the first time it sees a #.

Rather than mess with the toolchain, it's probably simpler to create a variant folder for the ZG, yes?

Reply:
I've got a commit ready to go if we are good with the ZG split out to a new variant folder.

Reply:
I will check that tomorrow.

Reply:
Ok this not work as the ld is not preprocess before.
I guess I should check if we could preprocess it before the link because it could be very useful for bootloader feature.

Reply:
I'll hold fire until you've looked into pre-processing.

Reply:
At a first step, I think my proposal is ok. I will study linker file preprocess for the bootloader management to avoid multiple linker file.

Reply:
Agreed - I've merged your changes... and I think it is automagically tacking into here....  (must... learn... github  WOW...)

Reply:
Note that I've a proposal to handle switch in linker script. ;)
See #404 
I've tested with https://github.com/fpistm/Arduino_Core_STM32/commit/f93d8f888c17fdea8dac8ec46a85dec33ceedd77

Reply:
Flash issue is handled.
LD prerprocess will be handle in an other PR.